src/protocols/secure_channel/CASESession.cpp

```diff
@@ -404,187 +445,283 @@ CHIP_ERROR CASESession::HandleSigmaR1(System::PacketBufferHandle && msg)
     ChipLogDetail(SecureChannel, "Peer assigned session key ID %d", initiatorSessionId);
     SetPeerSessionId(initiatorSessionId);
 
-    SuccessOrExit(err = tlvReader.Next());
-    VerifyOrExit(TLV::TagNumFromTag(tlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
-    SuccessOrExit(err = tlvReader.GetBytes(destinationIdentifier, sizeof(destinationIdentifier)));
+    if (sessionResumptionRequested && memcmp(resumptionID, mResumptionId, kCASEResumptionIDSize) == 0)
+    {
+        // Cross check resume1MIC with the shared secret
+        SuccessOrExit(err = ValidateSigmaResumeMIC(resume1MICSpan, ByteSpan(initiatorRandom), ByteSpan(mResumptionId),
```

Comment:
Failure to match resume MIC should not error, but actually fall-through to process a new session establishment.